From: Dima Kogan <dima@secretsauce.net>
Date: Sun, 18 Nov 2012 01:39:41 -0800
Subject: better test failure reporting if openssl not installed

---
 tests/test1.sh |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tests/test1.sh b/tests/test1.sh
index 4e3af05..4772495 100755
--- a/tests/test1.sh
+++ b/tests/test1.sh
@@ -37,10 +37,10 @@ checkmd5()
   fi
 
   md5val=`openssl md5 $1 | awk '{print $2;}'`
-  if [ $2 != $md5val ];
+  if [ x$2 != x$md5val ];
   then 
      echo failure:         $1
-     echo expected md5:    $2 "(got $md5val)"
+     echo expected md5:    $2 "(got '$md5val')"
      echo expected length: $3
      ls -l $1
      exit 1
