From 80086b39aec32222a763a90f68c2c824106ce4ef Mon Sep 17 00:00:00 2001
From: Simson Garfinkel <simsong@acm.org>
Date: Mon, 16 Sep 2013 11:17:21 -0400
Subject: [PATCH 07/17] moved } to outside #ifdef

---
 src/dfxml_writer.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: tcpflow/src/dfxml/src/dfxml_writer.cpp
===================================================================
--- tcpflow.orig/src/dfxml/src/dfxml_writer.cpp	2013-10-07 01:35:37.106522610 -0700
+++ tcpflow/src/dfxml/src/dfxml_writer.cpp	2013-10-07 01:35:37.106522610 -0700
@@ -437,9 +437,9 @@
     cpuid(0x80000006, &eax, &ebx, &ecx, &edx);
     xmlout("L1_cache_size", (int64_t) BFIX(ecx, 16, 31) * 1024);
     pop();
+}
 #undef BFIX
 #endif
-}
 
 void dfxml_writer::add_DFXML_execution_environment(const std::string &command_line)
 {
