From 7470ff50ba7a8af6bd45cc420d9b80d104c36721 Mon Sep 17 00:00:00 2001
From: Dima Kogan <dima@secretsauce.net>
Date: Tue, 5 Nov 2013 21:10:21 -0800
Subject: [PATCH] Setting ETHERTYPE_IPV6 and ETHERTYPE_VLAN explicitly to work
 on Debian/kfreebsd

---
 bulk_extractor_i.h | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

Index: tcpflow/src/be13_api/bulk_extractor_i.h
===================================================================
--- tcpflow.orig/src/be13_api/bulk_extractor_i.h	2013-10-07 01:35:37.106522610 -0700
+++ tcpflow/src/be13_api/bulk_extractor_i.h	2013-10-07 01:35:37.106522610 -0700
@@ -414,12 +414,20 @@ public:
 #define ETHERTYPE_REVARP        0x8035          /* Reverse ARP */
 #define ETHERTYPE_AT            0x809B          /* AppleTalk protocol */
 #define ETHERTYPE_AARP          0x80F3          /* AppleTalk ARP */
-#define ETHERTYPE_VLAN          0x8100          /* IEEE 802.1Q VLAN tagging */
 #define ETHERTYPE_IPX           0x8137          /* IPX */
-#define ETHERTYPE_IPV6          0x86dd          /* IP protocol version 6 */
 #define ETHERTYPE_LOOPBACK      0x9000          /* used to test interfaces */
 #endif
-    
+
+    // These two need to be checked separately on Debian/kfreebsd
+#ifndef ETHERTYPE_IPV6
+#define ETHERTYPE_IPV6          0x86dd          /* IP protocol version 6 */
+#endif
+
+#ifndef ETHERTYPE_VLAN
+#define ETHERTYPE_VLAN          0x8100          /* IEEE 802.1Q VLAN tagging */
+#endif
+
+
     inline u_short packet_info::nshort(const u_char *buf,size_t pos) 
     {
         return (buf[pos]<<8) | (buf[pos+1]);
-- 
1.8.4.rc3

