--- tcpflow-0.21.orig/config.guess
+++ tcpflow-0.21/config.guess
@@ -1,4 +1,9 @@
 #! /bin/sh
+
+if [ -x /usr/share/misc/config.guess ]; then
+    exec /usr/share/misc/config.guess "$@"
+fi
+
 # Attempt to guess a canonical system name.
 #   Copyright (C) 1992, 1993, 1994, 1995, 1996, 1997, 1998, 1999
 #   Free Software Foundation, Inc.
--- tcpflow-0.21.orig/config.sub
+++ tcpflow-0.21/config.sub
@@ -1,4 +1,9 @@
 #! /bin/sh
+
+if [ -x /usr/share/misc/config.sub ]; then
+    exec /usr/share/misc/config.sub "$@"
+fi
+
 # Configuration validation subroutine script, version 1.1.
 #   Copyright (C) 1991, 92-97, 1998, 1999 Free Software Foundation, Inc.
 # This file is (in principle) common to ALL GNU software.
